handy.h

```diff
@@ -2960,6 +2960,53 @@ last-inclusive range.
 
 #endif
 
+/* These are simple Marsaglia XOR-SHIFT RNG's for 64 and 32 bits. These
+ * RNG's are of reasonable quality, very fast, and have the interesting
+ * property that provided 'x' is non-zero they create a cycle of 2^32-1
+ * or 2^64-1 "random" like numbers, with the exception of 0. Thus they
+ * are very useful when you want an integer to "dance" in a random way,
+ * but you also never want it to become 0 and thus false.
+ *
+ * Obviously they leave x unchanged if it starts out as 0. 
+ *
+ * We have two variants just because that can be helpful in certain
+ * places. There is no advantage to either, they are equally bad as each
+ * other as far RNG's go.  The main point is they produce different 
+ * sequences.
+ *
+ * */
+
+#define PERL_XORSHIFT32_A(x)    \
+STMT_START {                    \
+    (x) ^= ((x) << 13);         \
+    (x) ^= ((x) >> 17);         \
+    (x) ^= ((x) << 5);          \
+} STMT_END
+
+/* 32 bit version */
+#define PERL_XORSHIFT64_A(x)    \
+STMT_START {                    \
+    (x) ^= ((x) << 13);         \
+    (x) ^= ((x) >> 7);          \
+    (x) ^= ((x) << 17);         \
+} STMT_END
+
+/* 32 bit version */
+#define PERL_XORSHIFT32_B(x)    \
+STMT_START {                    \
+    (x) ^= ((x) << 5);          \
+    (x) ^= ((x) >> 27);         \
+    (x) ^= ((x) << 8);          \
+} STMT_END
+
+#define PERL_XORSHIFT64_B(x)    \
```

Comment:
`PERL_XORSHIFT64_B()` is defined here but isn't used anywhere.

Should `PERL_SRAND_OVERRIDE_NEXT_CHILD()` in perl.h be using it on 64-bit platforms?

Reply:
I made the PL_srand_override be a U32 because most of the rest of the seeding logic seemed to use a U32.  So I am only using the _B variant of the 32 bit. I figured that if we supply a 32 bit _B variant "for completeness" we should supply a 64 bit as well even if we dont use it right now. I realize "completeness" is a threadbare excuse, if someone thought we shouldn't bother I would have no problem removing it.

Reply:
I enhanced the comments to explain, but as I said, im fine with removing the 64_B variant if you want.